### PR TITLE
feat(components): Normalize link Href

### LINKED ISSUE #633: Bodiless link always renders the url with trailing slash
## Description
The site link should render links in uniform pattern and consistent to help the platform team have consistent usage.

**Items discussed at standup:**

Bodiless link renders relative links with a trailing slash. 

Any absolute url's entered by content entry, should not add a trailing slash and be identical to value entered.
## Acceptance Criteria
AC1:
Bodiless link is configurable for

prepending a trailing slash,
appending a trailing slash,
removing a trailing slash or
do nothing
on relative URLs.

AC1a: If Prepending, the links are handled in this manner:

url without explicit host has leading slash added unless it begins with a ./
uls without explict host that start with ? are prepended with './' case: ?foo=bar -> ./?foo=bar
uls without explict host that start with # are not prepended: #foo -> #foo
AC1b: If Appending, the links are handled in this manner:

url with that is a link to a file should not have a trailing slash appended.

AC1c: For all cases, url with explicit host is not altered

AC2: Any normalization is applied when the link is saved.

AC3: Bodiless link defaults to strategy of prepending slash & adding append a trailing slash.

### PR DESCRIPTION
**Pull request by wodenx**
## Changes
- Added default link normalizer logic to bodiless link
- Use this logic in MenuLink to determine active page

## Related Issues
closes #633

### CODE REVIEW

File: examples/test-site/src/components/MegaMenu/SimpleMenu.token.tsx [hunk 1]
```diff
@@ -22,7 +23,14 @@ import { withDesign, addClasses } from '@bodiless/fclasses';
...
   Wrapper: addClasses('w-full bg-teal-600 text-white'),
   Item: addClasses('hover:bg-teal-500 min-w-100 leading-loose text-sm'),
-  Title: addClasses('block w-full px-3'),
+  Title: flow(
+    addClasses('block w-full px-3'),
+    withDesign({
+      Link: withDesign({
+        ActiveLink: asActiveMenuLink,
+      }),
+    }),
```
Comment (VancheeZze) [inline examples/test-site/src/components/MegaMenu/SimpleMenu.token.tsx:32]: Should we move this into its own token? `withBaseSubMenuStyles` also has it.

File: packages/bodiless-organisms/src/components/MainMenu/MenuLink.tsx [hunk 2]
```diff
@@ -29,30 +31,16 @@ const menulinkComponents: MenuLinkComponents = {
...
+const useIsCurrentPage = (href: string | undefined) => {
+  const { node } = useNode();
+  return new DefaultNormalHref(href).isSamePage(node.pagePath);
 };
 
 export type Props = {
   href?: string,
+  isCurrentPage?: boolean,
   unwrap?: Function,
+
```
Comment (VancheeZze) [inline packages/bodiless-organisms/src/components/MainMenu/MenuLink.tsx:43]: Extra line

**Review by VancheeZze — APPROVED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Comment by TatsianaMiraniuk** [comment]
Hi @wodenx ,

please see **issue_1**:
Relative to the page url started with . are not working:

**Steps to reproduce:**
1. In Edit Env, open /products/ page
2. switch to Edit mode
3. add a product card in the flow container
4. click an image placeholder to get Local Context Menu
5. click Edit link
6. add '.product-b' 
7. click v.
8. check the final link - switch to Static and click on the product card

**Expected results:**
/products/product-b/ opens

**Actual Results:**
/.product-b/ opens.

Per  description in Link form:
'Internal links beginning with a ‘.’ will be relative to the current page. Those not beginning with a ‘.’ will be prefixed with ‘/’ and be relative to the site root. '

**Comment by TatsianaMiraniuk** [comment]
> Hi @wodenx ,
> 
> please see **issue_1**:
> Relative to the page url started with . are not working:
> 
> **Steps to reproduce:**
> 
> 1. In Edit Env, open /products/ page
> 2. switch to Edit mode
> 3. add a product card in the flow container
> 4. click an image placeholder to get Local Context Menu
> 5. click Edit link
> 6. add '.product-b'
> 7. click v.
> 8. check the final link - switch to Static and click on the product card
> 
> **Expected results:**
> /products/product-b/ opens
> 
> **Actual Results:**
> /.product-b/ opens.
> 
> Per description in Link form:
> 'Internal links beginning with a ‘.’ will be relative to the current page. Those not beginning with a ‘.’ will be prefixed with ‘/’ and be relative to the site root. '

@wodenx, as discussed, since ./ is expected, can you please update description in Link form accordingly? 
'Internal links beginning with a ‘.’ will be relative to the current page. Those not beginning with a ‘.’ will be prefixed with ‘/’ and be relative to the site root. '

**Comment by TatsianaMiraniuk** [comment]
@wodenx ,

please see **issue_2**:

**Steps to Reproduce:**
1. open Edit env, /link-toggle/
2. Switch to Edit mode
3. in the strip slash, enter https://example.com
4. click Save
5. Reopen Link form

**Expected Results:**
https://example.com stays unchanged

**Actual Results:**
https://example.com/ shows

Issue is reproduced for both the current domain and for a different one.

**Review by VancheeZze — DISMISSED** [review]

**Comment by TatsianaMiraniuk** [comment]
@wodenx ,

can you please fix the following:
Screen Shot 2020-11-24 at 4 17 15 PM

Thanks!

**Review by VancheeZze — APPROVED** [review]

**Review by TatsianaMiraniuk — APPROVED** [review]

**Pull request merged**